[2025-06-02T14:15:38.170+0200] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-06-02T14:15:38.182+0200] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: macro_trading_dag.run_countercyclical_strategy manual__2025-06-02T12:15:15.689116+00:00 [queued]>
[2025-06-02T14:15:38.186+0200] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: macro_trading_dag.run_countercyclical_strategy manual__2025-06-02T12:15:15.689116+00:00 [queued]>
[2025-06-02T14:15:38.186+0200] {taskinstance.py:2303} INFO - Starting attempt 1 of 3
[2025-06-02T14:15:38.197+0200] {taskinstance.py:2327} INFO - Executing <Task(SparkSubmitOperator): run_countercyclical_strategy> on 2025-06-02 12:15:15.689116+00:00
[2025-06-02T14:15:38.206+0200] {standard_task_runner.py:90} INFO - Running: ['airflow', 'tasks', 'run', 'macro_trading_dag', 'run_countercyclical_strategy', 'manual__2025-06-02T12:15:15.689116+00:00', '--job-id', '82', '--raw', '--subdir', 'DAGS_FOLDER/macro_trading_dag.py', '--cfg-path', '/tmp/tmp4nbiw29p']
[2025-06-02T14:15:38.207+0200] {standard_task_runner.py:91} INFO - Job 82: Subtask run_countercyclical_strategy
[2025-06-02T14:15:38.208+0200] {logging_mixin.py:188} WARNING - /home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=93161) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2025-06-02T14:15:38.209+0200] {standard_task_runner.py:63} INFO - Started process 93181 to run task
[2025-06-02T14:15:38.246+0200] {task_command.py:426} INFO - Running <TaskInstance: macro_trading_dag.run_countercyclical_strategy manual__2025-06-02T12:15:15.689116+00:00 [running]> on host N15I711-16GR512.localdomain
[2025-06-02T14:15:38.307+0200] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='macro_trading_dag' AIRFLOW_CTX_TASK_ID='run_countercyclical_strategy' AIRFLOW_CTX_EXECUTION_DATE='2025-06-02T12:15:15.689116+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-06-02T12:15:15.689116+00:00'
[2025-06-02T14:15:38.308+0200] {taskinstance.py:430} INFO - ::endgroup::
[2025-06-02T14:15:38.311+0200] {base.py:84} INFO - Using connection ID 'spark_default' for task execution.
[2025-06-02T14:15:38.312+0200] {spark_submit.py:351} INFO - Spark-Submit cmd: spark-submit --master yarn --name arrow-spark --queue root.default /home/leoja/airflow/scripts/spark_strategy.py --input /home/leoja/airflow/data/combined_spreads.csv --output /home/leoja/airflow/data/results
[2025-06-02T14:15:39.513+0200] {spark_submit.py:521} INFO - 25/06/02 14:15:39 WARN Utils: Your hostname, N15I711-16GR512 resolves to a loopback address: 127.0.1.1; using 10.255.255.254 instead (on interface lo)
[2025-06-02T14:15:39.515+0200] {spark_submit.py:521} INFO - 25/06/02 14:15:39 WARN Utils: Set SPARK_LOCAL_IP if you need to bind to another address
[2025-06-02T14:15:39.566+0200] {spark_submit.py:521} INFO - Exception in thread "main" org.apache.spark.SparkException: When running with master 'yarn' either HADOOP_CONF_DIR or YARN_CONF_DIR must be set in the environment.
[2025-06-02T14:15:39.566+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.error(SparkSubmitArguments.scala:650)
[2025-06-02T14:15:39.566+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateSubmitArguments(SparkSubmitArguments.scala:281)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.validateArguments(SparkSubmitArguments.scala:237)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmitArguments.<init>(SparkSubmitArguments.scala:122)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2$$anon$3.<init>(SparkSubmit.scala:1108)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.parseArguments(SparkSubmit.scala:1108)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit.doSubmit(SparkSubmit.scala:86)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$$anon$2.doSubmit(SparkSubmit.scala:1125)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit$.main(SparkSubmit.scala:1134)
[2025-06-02T14:15:39.567+0200] {spark_submit.py:521} INFO - at org.apache.spark.deploy.SparkSubmit.main(SparkSubmit.scala)
[2025-06-02T14:15:39.591+0200] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-06-02T14:15:39.595+0200] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/leoja/airflow/airflow_venv/lib/python3.12/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 452, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master yarn --name arrow-spark --queue root.default /home/leoja/airflow/scripts/spark_strategy.py --input /home/leoja/airflow/data/combined_spreads.csv --output /home/leoja/airflow/data/results. Error code is: 1.
[2025-06-02T14:15:39.597+0200] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=macro_trading_dag, task_id=run_countercyclical_strategy, execution_date=20250602T121515, start_date=20250602T121538, end_date=20250602T121539
[2025-06-02T14:15:39.607+0200] {standard_task_runner.py:110} ERROR - Failed to execute job 82 for task run_countercyclical_strategy (Cannot execute: spark-submit --master yarn --name arrow-spark --queue root.default /home/leoja/airflow/scripts/spark_strategy.py --input /home/leoja/airflow/data/combined_spreads.csv --output /home/leoja/airflow/data/results. Error code is: 1.; 93181)
[2025-06-02T14:15:39.628+0200] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2025-06-02T14:15:39.640+0200] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2025-06-02T14:15:39.641+0200] {local_task_job_runner.py:222} INFO - ::endgroup::
